### PR TITLE
Make DisjunctionStrategy test deterministic

### LINKED ISSUE #1857: InExpressionRewriteRuleTest.DisjunctionStrategy fails
clang 7.0.0, apparently using `__GLIBCXX__`:

```
Differing subtrees
-------------- Actual LQP --------------
[0] [Predicate] col_a = 4
 \_[1] [MockNode 'Unnamed'] pruned: 0/2 columns

------------- Expected LQP -------------
[0] [Predicate] col_a = 3
 \_[1] [MockNode 'Unnamed'] pruned: 0/2 columns
-------------..............-------------
/home/moritz/Coding/hyrise/src/test/optimizer/strategy/in_expression_rewrite_rule_test.cpp:110: Failure
Failed
```

If the output of something is by nature indeterministic, maybe it is best to not test strict order. Maybe use visit_lqp to collect the nodes and assert that it's four UnionAll nodes followed by the predicates in any order (or something to that effect).

### PR DESCRIPTION
**Pull request by mrks**
fixes #1857

### CODE REVIEW

File: src/test/optimizer/strategy/in_expression_rewrite_rule_test.cpp [hunk 1]
```diff
@@ -80,34 +80,39 @@ TEST_F(InExpressionRewriteRuleTest, DisjunctionStrategy) {
...
+    // Checks that a given node is a predicate of the form `col_a = x` where x is an int and will be added to
+    // values_found_in_predicates
+    const auto verify_predicate_node = [&](const auto& node) {
+      EXPECT_EQ(node->type, LQPNodeType::Predicate);
+      auto predicate_node = std::dynamic_pointer_cast<PredicateNode>(node);
+      EXPECT_TRUE(predicate_node);
+      auto predicate = std::dynamic_pointer_cast<BinaryPredicateExpression>(predicate_node->predicate());
+      EXPECT_TRUE(predicate);
+      EXPECT_EQ(predicate->left_operand(), col_a);
+      EXPECT_EQ(predicate->right_operand()->type, ExpressionType::Value);
```
Comment (mrzzzrm) [inline src/test/optimizer/strategy/in_expression_rewrite_rule_test.cpp]: ASSERT_EQ, otherwise the next line would be executed (and result in UB, likely?)

Comment (mrks) [reply src/test/optimizer/strategy/in_expression_rewrite_rule_test.cpp]: Yepp. I honestly don't see too much of a value in `EXPECT_` anyway.

File: src/test/optimizer/strategy/in_expression_rewrite_rule_test.cpp [hunk 2]
```diff
@@ -80,34 +80,40 @@ TEST_F(InExpressionRewriteRuleTest, DisjunctionStrategy) {
     const auto input_lqp = PredicateNode::make(five_element_in_expression, node);
```
Comment (Bouncner) [inline src/test/optimizer/strategy/in_expression_rewrite_rule_test.cpp]: Not yours, but `five_element` misses an s.

Comment (mrks) [reply src/test/optimizer/strategy/in_expression_rewrite_rule_test.cpp]: I wouldn't say so. It is a five-element InExpression. Shouldn't matter, anyway.

**Pull request merged**